doc/api/stream.md

```diff
@@ -1590,6 +1596,10 @@ async function run() {
 run().catch(console.error);
 ```
 
+If any `Writable` or `Readable` stream emits `'close'` without being able to
+fully flush or drain, `stream.pipeline()` will error with
+`ERR_STREAM_PREMATURE_CLOSE`.
```

Comment:
This is a bit vague on purpose. Basically, the only thing that can be assumed is ERR_STREAM_PREMATURE_CLOSE will be emitted if something happened which caused data to be lost. Disregarding any specifics around event ordering and inspectable state.

Reply:
+1, ok.